contracts/lib/SignatureVerification.sol

```diff
@@ -36,110 +34,221 @@ contract SignatureVerification is SignatureVerificationErrors, LowLevelHelpers {
+            {
+                // Take the difference between the max ECDSA signature length
+                // and the actual signature length. Overflow desired for any
+                // values > 65. If the diff is not 0 or 1, it is not a valid
+                // ECDSA signature - move on to EIP1271 check.
+                let lenDiff := sub(ECDSA_MaxLength, signatureLength)
+
+                // Declare variable for recovered signer.
+                let recoveredSigner
+
+                // If diff is 0 or 1, it may be an ECDSA signature.
+                // Try to recover signer.
+                if iszero(gt(lenDiff, 1)) {
+                    // Read the signature `s` value.
+                    let originalSignatureS := mload(
+                        add(signature, ECDSA_signature_s_offset)
+                    )
+
+                    // Read the first byte of the word after `s`. If the
+                    // signature is 65 bytes, this will be the real `v` value.
+                    // If not, it will need to be modified - doing it this way
+                    // saves an extra condition.
+                    v := byte(
+                        0,
+                        mload(add(signature, ECDSA_signature_v_offset))
+                    )
+
+                    // If lenDiff is 1, parse 64-byte signature as ECDSA.
+                    if lenDiff {
+                        // Extract yParity from highest bit of vs and add 27 to
+                        // get v.
+                        v := add(
+                            shr(MaxUint8, originalSignatureS),
+                            Signature_lower_v
+                        )
+
+                        // Extract canonical s from vs, all but the highest bit.
+                        // Temporarily overwrite the original `s` value in the
+                        // signature.
+                        mstore(
+                            add(signature, ECDSA_signature_s_offset),
+                            and(
+                                originalSignatureS,
+                                EIP2098_allButHighestBitMask
+                            )
+                        )
+                    }
+                    // Temporarily overwrite the signature length with `v` to
+                    // conform to the expected input for ecrecover.
+                    mstore(signature, v)
+
+                    // Temporarily overwrite the word before the length with
+                    // `digest` to conform to the expected input for ecrecover.
+                    mstore(wordBeforeSignaturePtr, digest)
+
+                    // Attempt to recover the signer for the given signature. Do
+                    // not check the call status as ecrecover will return a null
+                    // address if the signature is invalid.
+                    pop(
+                        staticcall(
+                            Ecrecover_gas, // Gas used by ecrecover precompile.
```

Comment:
I think this is a bad idea. For one, this creates issues with repricing. Also, this code is more expensive at runtime as well as deploy time. Using `gas()` would be 1 gas cheaper at runtime and also saves 3 bytes at deploy time.

Reply:
Also I guess it's not like ecrecover can exhaust unnecessary gas anyway?

Reply:
see #500 @hrkrshnn